src/Cake.Core/IO/DirectoryPath.cs

```diff
@@ -51,6 +52,16 @@ public FilePath GetFilePath(FilePath path)
             return new FilePath(PathHelper.Combine(FullPath, path.GetFilename().FullPath));
         }
 
+        /// <summary>
+        /// Gets the directory path of a <see cref="DirectoryPath"/>.
+        /// </summary>
+        /// <param name="path">The path.</param>
+        /// <returns>A <see cref="DirectoryPath"/> to the parent directory of the given <see cref="DirectoryPath"/>.</returns>
+        public DirectoryPath GetParent()
+        {
+            return new DirectoryPath(new DirectoryInfo(FullPath).Parent.FullName);
```

Comment:
It looks like if there's no parent, the user would get an exception - I don't think that's the behavior we want. Perhaps it's better to return `null` in that case? (we'll need tests for that too)

Reply:
I don't see why not :)